### PR TITLE
UState.universe_decl contains levels not ids

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by gares** [comment]
I understand the new types, and it clearly better suits my use of it in Coq-Elpi.
But I've no idea if it also makes more sense for Coq itself, I'll let @coq/universes-maintainers speak. @mattam82 WDYT?

File: engine/uState.ml
Comment (SkySkimmer) [inline engine/uState.ml:461]: IMO it was strange to have half of it using ids and half real levels (in the constraints)

**Review by ppedrot — APPROVED** [review]
The new API seems more natural to me as well. We shouldn't have had a mix between constraints and names indeed.

**Comment by ppedrot** [comment]
Since nobody seems to react, I'm assigning and will merge soon if nobody complains.

**Comment by gares** [comment]
I'm realizing that without this one adding decent support for unverse polymorphism to elpi is quite hard. I could not find a way to implement this "on my side" waiting for 8.17. This is an API change, but not impacting clients according to CI...

**Comment by gares** [comment]
I found another way to skin the cat, which  is to export an API in UState that names a universe level.

With that one, I think I'm good (currently I have to obj magic UState.t to something I can act on, which is not nice, but I can live with it).

**Comment by gares** [comment]
If you that API is reasonable I can prepare a PR and in that case it can't break any client since the function is new and hence safer to backport. But I can also live with the obj magic...

**Comment by SkySkimmer** [comment]
What's wrong with this PR?

**Comment by Alizter** [comment]
@SkySkimmer I reset the refman job for this PR not realising it was too old. Sorry about that.

**Comment by gares** [comment]
Nothing is wrong other than it is not in 8.16 :-)

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**
